pymodbus/transport/transport.py

```diff
@@ -177,36 +181,32 @@ def __init__(
             host = self.comm_params.host
             port = int(self.comm_params.port)
         if self.comm_params.comm_type == CommType.SERIAL:
-            host, port = self.init_setup_serial(host, port)
-            if not host and not port:
+            if NULLMODEM_HOST in host:
```

Comment:
This doesn't make sense to me...you split something that will be split later ?

Reply:
It didn't make sense to me either, but that's how the existing code works.

I suspect there is further improvement possible here.

Reply:
Split out into https://github.com/pymodbus-dev/pymodbus/pull/1866